llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -808,8 +807,7 @@ Value *VPInstruction::generate(VPTransformState &State) {
         cast<StructType>(State.TypeAnalysis.inferScalarType(getOperand(0)));
     Value *Res = PoisonValue::get(toVectorizedTy(StructTy, State.VF));
     for (const auto &[LaneIndex, Op] : enumerate(operands())) {
-      for (unsigned FieldIndex = 0; FieldIndex != StructTy->getNumElements();
-           FieldIndex++) {
+      for (unsigned FieldIndex : seq<unsigned>(StructTy->getNumElements())) {
```

Comment:
This works fine, but I think this is just personal preference. I don't find the new version any more readable than the old one to be honest.

Reply:
seq has the advantage that getNumElements() is only evaluated once, and the iota_range is expanded -- this would be equivalent to `I = Begin, E = End; I != E; ++I`, and hence more efficient.